### PR TITLE
Mat-4292: Shared header, move create dialog, setup for system module.

### PR DESCRIPTION
**Pull request by mcmcphillips**
## MADiE PR

Jira Ticket: [MAT-4292](https://jira.cms.gov/browse/MAT-4292)
(Optional) Related Tickets:
This PR depends on the following PRs:
 [madie-measure](https://github.com/MeasureAuthoringTool/madie-measure/pull/97)
 [madie-root](https://github.com/MeasureAuthoringTool/madie-root/pull/82)
 [madie-util](https://github.com/MeasureAuthoringTool/madie-util/pull/1)

This PR:
-  updates implements a shared header that tracks state across micro frontends.
-  subscribes to an rjsx store for updates from measures based on measureContext.
-  updates existing tests, moves measureCreation and dialog to layout since it's no longer accessed from Measure.
-  emits a custom event on successful measure creation to allow measures to know when to retrieve new items from the landing page.
- Adds typing, jest configuration, and mocks required to use madie-utility.
### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included
* [x] No extemporaneous files are included (i.e Complied files or testing results)
* [x] This PR is into the **correct branch**.
* [x] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [x] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependencies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/components/NewMeasure/CreateNewMeasureDialog.tsx [hunk 1]
```diff
@@ -0,0 +1,556 @@
...
+                      inputProps={{
+                        datatestId: "auto-generate-checkbox-input",
+                      }}
+                      // onChange={(e) => {
+                      //   const val = e.target.checked;
+                      //   if (val && manualId) {
+                      //     setManualId(false);
+                      //   }
+                      //   setAutoGenerate(e.target.checked);
+                      // }}
```
Comment (jkotanchik-SB) [inline src/components/NewMeasure/CreateNewMeasureDialog.tsx:385]: Dead code or saving for later?

File: src/components/NewMeasure/bulkCreate.tsx [hunk 2]
```diff
@@ -0,0 +1,67 @@
+import { MeasureScoring } from "@madie/madie-models/dist/MeasureScoring";
+import { Measure } from "@madie/madie-models/dist/Measure";
+import { Model } from "@madie/madie-models/dist/Model";
+// This is some frontend logic for adding a bunch of measures
+// use Bulk create in createNewMeasure as a button press for a quick load and db crash
```
Comment (jkotanchik-SB) [inline src/components/NewMeasure/bulkCreate.tsx:5]: 🤣

File: src/components/PageHeader/PageHeader.tsx [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+import "twin.macro";
+import "styled-components/macro";
+import "./pageHeader.scss";
+
+const PageHeader = () => {
+  const { pathname } = useLocation();
+  // homepage stuff
+  const [createOpen, setCreateOpen] = useState<boolean>(false);
+  const [measureState, setMeasureState] = useState<any>(measureStore.state);
+  //
```
Comment (jkotanchik-SB) [inline src/components/PageHeader/PageHeader.tsx]: Remove or populate the empty comment pls.

File: src/models/MeasureSchemaValidator.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import * as Yup from "yup";
+import { Model } from "@madie/madie-models/dist/Model";
+import { MeasureScoring } from "@madie/madie-models/dist/MeasureScoring";
+
+export const MeasureSchemaValidator = Yup.object().shape({
+  measureName: Yup.string()
+    .max(500, "A measure name cannot be more than 500 characters.")
+    .required("A measure name is required.")
+    .matches(/[a-zA-Z]/, "A measure name must contain at least one letter.")
+    .matches(/^((?!_).)*$/, "Measure Name must not contain '_' (underscores)."),
```
Comment (jkotanchik-SB) [inline src/models/MeasureSchemaValidator.ts:10]: Minor editorial request: Do you mind making "Measure Name" have consistent captialization in these validation messages, please?

File: src/models/MeasureSchemaValidator.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      /^((?!_).)*$/,
+      "Measure library name must not contain '_' (underscores)."
+    )
+    .matches(
+      /^[A-Z][a-zA-Z0-9]*$/,
+      "Measure library name must start with an upper case letter, followed by alpha-numeric character(s) and must not contain spaces or other special characters."
+    ),
+  measureScoring: Yup.string()
+    .oneOf(Object.values(MeasureScoring))
+    .required("Measure Scoring is required."),
```
Comment (jkotanchik-SB) [inline src/models/MeasureSchemaValidator.ts]: Minor editorial request: Do you mind making "Measure *" have consistent capitalization in these validation messages, please? 
Measure name -> Measure Name, Measure library -> Measure Library, etc etc.

**Review by jkotanchik-SB — DISMISSED** [review]

**Review by gregory-akins — APPROVED** [review]
